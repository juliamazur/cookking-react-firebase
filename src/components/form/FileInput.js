import React from 'react';

import FormControl from '@material-ui/core/FormControl';

class FileInput extends React.Component {

  render() {
    return (
      <FormControl fullWidth={true}>
        <input
          data-test="fileInput"
          ref={this.props.reference}
          accept="image/*"
          id="button-file"
          type="file"
          style={{ display: 'none' }}
          onChange={this.props.handleChange}
        />
      </FormControl>
    );
  }
}

export default FileInput;
