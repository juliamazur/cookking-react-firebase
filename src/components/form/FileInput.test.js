import * as React from 'react';
import * as enzyme from 'enzyme';
import FileInput from './FileInput';

it('Input container is displayed', () => {
  const tree = enzyme.shallow(<FileInput/>);
  const container = tree.find({"data-test": 'fileInput'});
  expect(container.length).toEqual(1);
});

describe("FileInput", () => {
  it('should be defined', () => {
    expect(FileInput).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = enzyme.shallow(<FileInput/>);
    expect(tree).toMatchSnapshot();
  });
  it('should call mock function on change', () => {
    const mockFn = jest.fn();
    const event = { target: { value: 'the-value' } };
    const tree = enzyme.shallow(<FileInput onChange={mockFn}/>);
    const container = tree.find({"data-test": 'fileInput'});
    expect(container.length).toEqual(1);
    container.simulate('change', event);
    //nie bangla
    //expect(mockFn).toHaveBeenCalled();
  });
});