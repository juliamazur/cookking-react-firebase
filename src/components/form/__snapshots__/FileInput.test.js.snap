// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInput should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FileInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <input
        accept="image/*"
        data-test="fileInput"
        id="button-file"
        onChange={undefined}
        type="file"
/>,
      "fullWidth": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "accept": "image/*",
        "data-test": "fileInput",
        "id": "button-file",
        "onChange": undefined,
        "type": "file",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <input
          accept="image/*"
          data-test="fileInput"
          id="button-file"
          onChange={undefined}
          type="file"
/>,
        "fullWidth": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "accept": "image/*",
          "data-test": "fileInput",
          "id": "button-file",
          "onChange": undefined,
          "type": "file",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
